security/s2a/internal/crypter/expander_test.go

```diff
@@ -0,0 +1,66 @@
+package crypter
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"google.golang.org/grpc/security/s2a/internal/crypter/testutil"
+	"testing"
+)
+
+type hkdfTestVector struct {
+	comment        string
+	prk, info, okm []byte
+	length         int
+}
+
+func TestExpanderExpansionInvalidLength(t *testing.T) {
+	tc := hkdfTestVector{
+		comment: "invalid length",
+		prk:     testutil.Dehex("07770932df0ddc3f0dc47bba6390b6c73bb50f9c3122ec844ad7c2b3e5"),
+		info:    testutil.Dehex("f0f1f2f3f4f5f6f7f8f9"),
+		length:  10000,
+	}
+
+	expander := newDefaultHKDFExpander()
+	got, err := expander.expand(sha256.New, tc.prk, tc.info, tc.length)
+	if err == nil {
+		t.Errorf("expand(sha256.New, %v, %v, %v) = %v, expected error but no error returned", tc.prk, tc.info, tc.length, got)
+	}
+}
+
+func TestExpanderExpansion(t *testing.T) {
```

Comment:
If a test is testing a specific function, it's a good to call the test `TestFuncName`. In this case, this can be called `TestExpand`.

Reply:
Done.